File: adventofcode-2016/src/main/java/nl/mvdr/adventofcode/adventofcode2016/day11/Part1.java
package nl.mvdr.adventofcode.adventofcode2016.day11;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import nl.mvdr.adventofcode.solver.Solver;

public class Part1 implements Solver {

    private static final Logger LOGGER = LoggerFactory.getLogger(Part1.class);
    
    @Override
    public String solve(String inputfile) {
        return new RadioisotopeThermoelectricGenerators(false).solve(inputfile);
    }
    
    public static void main(String[] args) {
        Part1 instance = new Part1();

        String result = instance.solve("input-day11.txt");

        LOGGER.info(result);
    }
}

File: adventofcode-2025/src/main/kotlin/nl/mvdr/adventofcode/adventofcode2025/day11/Part1.kt
package nl.mvdr.adventofcode.adventofcode2025.day11

import io.github.oshai.kotlinlogging.KotlinLogging
import nl.mvdr.adventofcode.solver.FunctionSolver

private val logger = KotlinLogging.logger{}

fun solvePart1(lines: Sequence<String>) = createAlgorithm(lines)
    .getAllPaths("you", "out")
    .size

fun main() {
    val result = FunctionSolver(::solvePart1).solve("input-day11.txt")
    logger.info { result }
}
